src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2146,26 +2154,37 @@ void Session::send_json_error_message()
     REALM_ASSERT(!m_unbind_message_sent);
     REALM_ASSERT(m_error_to_send);
     REALM_ASSERT(m_client_error);
+    REALM_ASSERT(!m_wait_for_error_timer);
 
     ClientProtocol& protocol = m_conn.get_client_protocol();
     OutputBuffer& out = m_conn.get_output_buffer();
     session_ident_type session_ident = get_ident();
-    std::error_code ec = m_client_error->code();
+    auto client_error = m_client_error->code();
+    auto protocol_error = static_cast<int>(client_error_to_protocol_error(client_error));
```

Comment:
can we do the static cast in the logging statement rather than in the variable assignment?